app/models/concerns/has_wise_recipient.rb

```diff
@@ -0,0 +1,161 @@
+# frozen_string_literal: true
+
+module HasWiseRecipient
+  extend ActiveSupport::Concern
+
+  included do
+    include CountryEnumable
+    has_country_enum(field: :recipient_country)
+
+    # validate do
+    #   unless bic_code.nil? || bic_code.match(/[A-Z]{4}([A-Z]{2})[A-Z0-9]{2}([A-Z0-9]{3})?$/) # https://www.johndcook.com/blog/2024/01/29/swift/
+    #     errors.add(:bic_code, "is not a valid SWIFT / BIC code")
+    #   end
+    # end
+
+    # validate do
+    #   if IBAN_FORMATS[bank_country.to_sym] && !account_number.match(IBAN_FORMATS[bank_country.to_sym])
+    #     errors.add(:account_number, "does not meet the required format for this country")
+    #   end
+    # end
+
+    validate do
+      if POSTAL_CODE_FORMATS[recipient_country.to_sym] && !address_postal_code.match(POSTAL_CODE_FORMATS[recipient_country.to_sym])
+        errors.add(:address_postal_code, "does not meet the required format for this country")
+      end
+    end
+
+    validate do
+      unless currency.in?(AVAILABLE_CURRENCIES)
+        errors.add(:currency, "is not supported for Wise transfers")
+      end
+    end
+
+    # the SWIFT messaging system supports a very limited set of characters.
+    # https://column.com/docs/international-wires/#valid-characters-permitted
+
+    # validate do
+    #   error = "contains invalid characters; the SWIFT system only supports the English alphabet and numbers."
+    #   regex = /[^A-Za-z0-9\-?:( ).,'+\/]/
+
+    #   errors.add(:address_line1, error) if address_line1.match(regex)
+    #   errors.add(:address_line2, error) if address_line2.present? && address_line2.match(regex)
+    #   errors.add(:address_postal_code, error) if address_postal_code.match(regex)
+    #   errors.add(:address_state, error) if address_state.match(regex)
+
+    #   Wire.recipient_information_accessors.excluding("legal_type", "email").each do |recipient_information_accessor|
+    #     errors.add(recipient_information_accessor, error) if recipient_information[recipient_information_accessor]&.match(regex)
+    #   end
+    # end
+
+    # see https://column.com/docs/api/#counterparty/create for valid options, under "legal_type"
+
+    # View https://github.com/hackclub/hcb/issues/9037 for context. Limited in India only, at the moment.
+
+    validate on: :create do
+      if recipient_information[:purpose_code].present? && RESTRICTED_PURPOSE_CODES[recipient_country.to_sym]&.include?(recipient_information[:purpose_code])
+        errors.add(:purpose_code, "can not be used on HCB, please use a more specific purpose code or contact us.")
+      end
+    end
+
+    def self.information_required_for(currency) # country can be null, in which case, only the general fields will be returned.
```

Comment:
Would be good to include a comment about where this information came from + how to extend this